GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,619 @@
+
+*Note: from the UI, for urgent proposals we should maybe add a ‘Not Urgent’ option that casts a `NoWithVeto` vote.*
+
+### Quorum 
+
+Quorum is defined as the minimum percentage of voting power that needs to be casted on a proposal for the result to be valid. 
+
+In the initial version of the governance module, there will be no quorum enforced by the protocol. Participation is ensured via the combination of inheritance and validator's punishment for non-voting.
+
+### Threshold
+
+Threshold is defined as the minimum ratio of `Yes` votes to `No` votes for the proposal to be accepted.
+
+Initially, the threshold is set at 50% with a possibility to veto if more than 1/3rd of votes (excluding `Abstain` votes) are `NoWithVeto` votes. This means that proposals are accepted is the ratio of `Yes` votes to `No` votes at the end of the voting period is superior to 50% and if the number of `NoWithVeto` votes is inferior to 1/3rd of total votes (excluding `Abstain`).
+
+`Urgent` proposals also work with the aforementioned threshold, except there is another condition that can accelerate the acceptance of the proposal. Namely, if the ratio of `Yes` votes to `InitTotalVotingPower` exceeds 2/3, `UrgentProposal` will be immediately accepted, even if the `Voting period` is not finished. `InitTotalVotingPower` is the total voting power of all bonded Atom holders at the moment when the vote opens.
+
+### Inheritance
+
+If a delegator does not vote, it will inherit its validator vote.
+
+- If the delegator votes before its validator, it will not inherit from the validator's vote.
+- If the delegator votes after its validaotor, it will override its validator vote with its own vote. If the proposal is a `Urgent` proposal, it is possible that the vote will close before delegators have a chance to react and override their validator's vote. This is not a problem, as `Urgent` proposals require more than 2/3rd of the total voting power to pass before the end of the voting period. If more than 2/3rd of validators collude, they can censor the votes of delegators anyway. This is an inherent limit of byzantine fault tolerant systems. 
+
+### Validator’s punishment for non-voting
+
+Validators are required to vote on all proposals to ensure that results have legitimacy. Voting is part of validators' directives and failure to do it will result in a penalty. 
+
+If a validator’s address is not in the list of addresses that voted on a proposal and if the vote is closed (i.e. `MinDeposit` was reached and `Voting period` is over), then any user of the Hub can submit a `TxGovernancePenalty` that will result in a partial slash `GovernancePenalty` of the validator's stake. The user that submitted the `TxGovernancePenalty` transaction will receive `GovernanceReward` from the `GovernancePenalty`.
```

Comment:
Why have a user submit a transaction. Just penalise the validator automatically.

Reply:
Feels computationally expensive to me. I have Ethereum reflexes, to me state mutation should always (or almost) be triggered by users sending a tx and paying the associated fee.
I'm not against automatic punishment but computational cost needs to be evaluated.

Reply:
@rigelrozanski

Reply:
I think iterating through the validator set for each governance proposal is not a problem... Iterating over the validator set is only really a problem if it needs to happen every block, given that proposals are infrequent relatively speaking we should be OK to iterate through the validator set. 

So in line with adrians thinking I'd edit to not include a Tx for punishment here

Reply:
Hmm ok but is it possible to trigger the iteration exactly `VotingPeriod` blocks after the proposal reaches `MinDeposit`. Like when `MinDeposit` is reached the app will perform the iteration in `BeginBlock` 2 weeks after, without iterating over all proposals?

Reply:
It's possible, when adding a deposit you check is `MinDeposit` is reached, if it is, then you add the proposal ID to a `ProposalProcessingQueue` where the oldest element is checked each round... this is a similar to unbonding for the staking module, we can reuse the queue

Reply:
Ok, I'll modify the spec accordingly then